backend/src/layer/logic/mealplan_management/relation_resolver.rs

```diff
@@ -0,0 +1,254 @@
+use crate::interface::mensa_parser::model::{Dish, ParseCanteen, ParseLine};
+use crate::interface::persistent_data::model::Line;
+use crate::interface::persistent_data::{DataError, MealplanManagementDataAccess};
+use crate::util::Date;
+use std::slice::Iter;
+use tracing::warn;
+
+pub struct RelationResolver<DataAccess>
+where
+    DataAccess: MealplanManagementDataAccess + Send + Sync,
+{
+    db: DataAccess,
+}
+
+impl<DataAccess> RelationResolver<DataAccess>
+where
+    DataAccess: MealplanManagementDataAccess + Send + Sync,
+{
+    pub const fn _new(db: DataAccess) -> Self {
+        Self { db }
+    }
+
+    const PERCENTAGE: f64 = 0.8;
+    const fn get_edge_case_meal() -> &'static str {
+        "je 100 g"
+    }
+
+    /// This method resolves relation problems with canteen data and the corresponding database.<br>
+    /// After each resolve the object gets injected into the database.<br>
+    /// If an similar object already exists, the existing object will be updated with the new object data.<br>
+    /// `canteen: ParseCanteen`<br>This struct contains all canteen data e.g. lines and dishes.<br>
+    /// `date: Date`<br>This date decides when the meal will be served next.<br>
+    /// **Return**<br>Occurring errors get passed to the `MealPlanManger`.
+    pub async fn resolve(&self, canteen: ParseCanteen, date: Date) -> Result<(), DataError> {
+        match self.db.get_similar_canteen(&canteen.name).await? {
+            Some(similar_canteen) => {
+                self.db
+                    .update_canteen(similar_canteen.id, &canteen.name)
+                    .await?
+            }
+            None => self.db.insert_canteen(&canteen.name).await?,
+        };
+
+        for line in canteen.lines {
+            let name = &line.name.clone();
+            if let Err(_e) = self.handle_line(line, date).await {
+                warn!("Skip line '{:?}' as it could not be resolved", name);
```

Comment:
do you really need to debug print the name?

Reply:
I think this is useful for failure determination.

Reply:
yea, but can't you print it just normal with `{}` (without `:?`?